blockmanager.go

```diff
@@ -992,8 +1016,8 @@ func (b *blockManager) fetchHeaderBlocks() {
 			continue
 		}
 		if !haveInv {
-			b.requestedBlocks[*node.hash] = struct{}{}
```

Comment:
This entry in the server-wide map should not have been removed.  Otherwise, the check to see if the block has already been requested by another peer will fail.